DNB.js

```diff
@@ -0,0 +1,346 @@
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2015 Philipp Zumstein
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+var typeMapping = {
+	//"Blindendrucke" 
+	"Bücher" : "book",
+	//"Elektronische Datenträger" 
+	"Filme/Hörbücher" : "videoRecording",
+	"Karten" : "map",
+	//"Medienkombinationen"
+	//"Mikroformen"
+	"Musiktonträger" : "audioRecording",
+	//"Musiknoten" 
+	"Artikel" : "journalArticle",
+	//"Online Ressourcen"
+	//"Zeitschriften/Serien"
+	//"Ausgaben/Hefte"
+	"archivierte Webseiten" : "webpage",
+	//"Gesamttitel Sammlung/Nachlass"
+	"Manuskripte" : "manuscript",
+	"Briefe" : "letter",
+	//"Lebensdokumente"
+	//"Sammlungen"
+	//"Trägermaterialien (Papiere und Einbände)"
+	"Bilder/Grafiken" : "artwork"
+	//"Flugblätter"
+};
+
+function detectWeb(doc, url) {
+	if (
+		url.indexOf('method=showFullRecord')>-1 ||
+		(url.indexOf('method=simpleSearch')>-1 && doc.getElementById('fullRecordTable'))
+	) {
+		var type=ZU.xpathText(doc, '//table[@id="fullRecordTable"]/tbody/tr/td/img/@alt');
+		if (typeMapping[type]) {
+			return typeMapping[type];
+		} else {
+			return "book";
+		}
+	} else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//table[@id="searchresult"]//a[contains(@id, "recordLink")]');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].firstChild.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = new Array();
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc, url) {
+	var marc = ZU.xpath(doc, '//div[@class="link"]//a[contains(@href, "/about/marcxml")]');
+	if (marc.length) {
+		var marcUrl = marc[0].href;
+		
+		ZU.doGet(marcUrl, function(result) {
+			//the first line is empty and has to be deleted
+			result = result.replace(/^\s*\n\s*<record/, '<record');
```

Comment:
Fixed in https://github.com/zotero/translators/commit/704e02e060bd68ce6d9a1f22202525c6f793f101 No longer necessary